src/storage/InMemoryResourceStore.ts

```diff
@@ -147,13 +148,19 @@ export class InMemoryResourceStore implements ResourceStore {
    * @returns The resulting Representation.
    */
   private async generateRepresentation(path: string): Promise<Representation> {
+    // Note: when converting to a complete ResourceStore and using readable-stream
+    // object mode should be set correctly here (now fixed due to Node 10)
     const source = this.store[path];
-    const arr = await arrayifyStream(source.data);
-    source.data = streamifyArray([ ...arr ]);
+    const streamInternal = new PassThrough({ writableObjectMode: true, readableObjectMode: true });
+    const streamOutput = new PassThrough({ writableObjectMode: true, readableObjectMode: true });
+    source.data.pipe(streamInternal);
+    source.data.pipe(streamOutput);
```

Comment:
No error events will be forwarded here, but I suspect this is not a problem because error checking would already have been done before?

Reply:
True, we should actually use the the util function of #77 here once that gets merged. But I'll add the manual error handling for now, since there might be cases I'm not thinking of that could throw errors.

Reply:
No I just realized that you're right and this can never happen since it would have triggered when adding the resource to the store initially, so there's no need (discovered this when I realized it was impossible to test :D ).